src/allocation_engine/interval_tree.rs

```diff
@@ -142,6 +142,16 @@ pub enum NodeState {
     Allocated,
 }
 
+impl NodeState {
+    fn replace(&mut self, value: NodeState) -> Self {
```

Comment:
How would this be used? Since it's an enum, why not just set another value directly?